torch/export/_trace.py

```diff
@@ -1425,10 +1151,322 @@ def forward(self, *args, **kwargs):
+                        for key, val in pytree.tree_map(
+                            strip_root, nn_module_stack
+                        ).items()
+                    }
+
+            return gm, sig
+
+        return _aot_export_non_strict
+
+    (
+        fake_mode,
+        fake_args,
+        fake_kwargs,
+        equalities_inputs,
+        original_signature,
+    ) = make_fake_inputs(
+        mod, args, kwargs, dynamic_shapes, _is_torch_jit_trace=_is_torch_jit_trace
+    )
+
+    fake_params_buffers = make_fake_params_buffers(fake_mode, _get_params_buffers(mod))
+
+    with fake_mode:
+        with _fakify_script_objects(mod, fake_args, fake_kwargs, fake_mode) as (
+            patched_mod,
+            new_fake_args,
+            new_fake_kwargs,
+            new_fake_constant_attrs,
+            map_fake_to_real,
+        ):
+            aten_export_artifact = _export_to_aten_ir(
+                patched_mod,
+                new_fake_args,
+                new_fake_kwargs,
+                fake_params_buffers,
+                new_fake_constant_attrs,
+                pre_dispatch=pre_dispatch,
+                transform=_tuplify_outputs,
+                should_insert_runtime_assertion=True,
+                _is_torch_jit_trace=_is_torch_jit_trace,
+            )
+            # aten_export_artifact.constants contains only fake script objects, we need to map them back
+            aten_export_artifact.constants = {
+                fqn: map_fake_to_real[obj] if isinstance(obj, FakeScriptObject) else obj
+                for fqn, obj in aten_export_artifact.constants.items()
+            }
+
+    aten_export_artifact.gm.meta["inline_constraints"] = {
+        k: v
+        for k, v in fake_mode.shape_env.var_to_range.items()
+        if free_unbacked_symbols(k)
+    }
+    num_lifted = len(
+        [
+            spec
+            for spec in aten_export_artifact.sig.input_specs
+            if spec.kind != InputKind.USER_INPUT
+        ]
+    )
+    try:
+        produce_guards_and_solve_constraints(
+            fake_mode,
+            aten_export_artifact.gm,
+            equalities_inputs,
+            original_signature,
+            _disable_forced_specializations=_disable_forced_specializations,
+            _is_torch_jit_trace=_is_torch_jit_trace,
+        )
+    except (ConstraintViolationError, ValueRangeError) as e:
+        raise UserError(UserErrorType.CONSTRAINT_VIOLATION, str(e))  # noqa: B904
+
+    combined_args = _combine_args(mod, args, kwargs)
+    range_constraints = make_constraints(
+        fake_mode,
+        aten_export_artifact.gm,
+        combined_args,
+        dynamic_shapes,
+        num_lifted,
+    )
+
+    assert out_spec is not None
+
+    gm = aten_export_artifact.gm
+
+    gm.meta["forward_arg_names"] = forward_arg_names
+    module_call_signatures = {
+        strip_root(fqn): ModuleCallSignature(inputs=[], outputs=[], **specs)
+        for fqn, specs in module_call_specs.items()
+    }
```

Comment:
Maybe we can share this code between strict/nonstrict by moving this, and the `CollectTracepointPass` call to the toplevel `_export` call. non-strict can run the pass on line 1323 before returning. Then we can get rid of the `module_call_signatures` output in `_strict_export` and `_non_strict_export`